gateway/server.go

```diff
@@ -224,3 +225,39 @@ func main() {
 
 	log.Fatal(s.ListenAndServe())
 }
+
+//runMetricsServer Listen on a separate HTTP port for Prometheus metrics to keep this accessible from
+// the internal network only.
+func runMetricsServer() {
+	metricsHandler := metrics.PrometheusHandler()
+	router := mux.NewRouter()
+	router.Handle("/metrics", metricsHandler)
+	router.HandleFunc("/healthz", healthzHandler)
+
+	port := 8082
+	readTimeout := time.Duration(5) * time.Second
+	writeTimeout := time.Duration(5) * time.Second
+
+	s := &http.Server{
+		Addr:           fmt.Sprintf(":%d", port),
+		ReadTimeout:    readTimeout,
+		WriteTimeout:   writeTimeout,
+		MaxHeaderBytes: http.DefaultMaxHeaderBytes,
+		Handler:        router,
+	}
+
+	log.Fatal(s.ListenAndServe())
+}
+
+func healthzHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Do we already have a healthz handler for the gateway itself? Perhaps this handler should be moved to the `handlers` package or shared?

Reply:
gateway using reverse proxy handler for `healthzHandler` as well. Shouldn't this have different healthz handler?